curses.h

```diff
@@ -83,11 +88,15 @@ extern "C"
 typedef unsigned char bool;    /* PDCurses Boolean type */
 
 #ifdef CHTYPE_LONG
-# if _LP64
-typedef unsigned int chtype;
-# else
-typedef unsigned long chtype;  /* 16-bit attr + 16-bit char */
-# endif
+    #if(CHTYPE_LONG >= 2)       /* "non-standard" 64-bit chtypes     */
+        typedef uint64_t chtype;
+    #else                        /* "Standard" CHTYPE_LONG case,  32-bit: */
+        # if _LP64
+            typedef unsigned int chtype;
+        # else
+            typedef unsigned long chtype;  /* 16-bit attr + 16-bit char */
```

Comment:
Hi Luke,

On 2016-03-23 00:27, Luke-Jr wrote:
> In curses.h <https://github.com/wmcbrine/PDCurses/pull/6#discussion_r57108369>:
>
>> @@ -83,11 +88,15 @@ extern "C"
>>  typedef unsigned char bool;    /* PDCurses Boolean type */
>>
>>  #ifdef CHTYPE_LONG
>> -# if _LP64
>> -typedef unsigned int chtype;
>> -# else
>> -typedef unsigned long chtype;  /* 16-bit attr + 16-bit char */
>> -# endif
>> +    #if(CHTYPE_LONG >= 2)       /* "non-standard" 64-bit chtypes     */
>> +        typedef uint64_t chtype;
>> +    #else                        /* "Standard" CHTYPE_LONG case,  32-bit: */
>> +        # if _LP64
>> +            typedef unsigned int chtype;
>> +        # else
>> +            typedef unsigned long chtype;  /* 16-bit attr + 16-bit char */
>
> You're including stdint.h, so why not use uint32_t?

    Um.  Looks to me as if you have a good point there.  I'd prefer letting
the folks who write 'stdint.h' puzzle out which size is which.  Figuring
out what size various types are isn't as simple as checking _LP64;  there
are different macros to check on different compilers,  and there's a reason
'stdint.h' is often a complicated mess.  It could be we have people out
there thinking they're getting 32-bit chtypes when they're really getting
64-bit chtypes.

    Similarly,  a few lines further down :

typedef unsigned short chtype; /* 8-bit attr + 8-bit char */

    should,  I think,  use uint16_t.  I'll make that change.

    (I didn't do this before,  due to tunnel vision.  I was adding my code
for 64-bit chtypes and didn't look too closely at the existing code
for 32-bit and 16-bit chtypes.)

-- Bill

> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/wmcbrine/PDCurses/pull/6/files/c1ab1e40dc35a905edd65e5263ce9d7029d5e452#r57108369>
>